tests/sales/sales_test.go

```diff
@@ -1019,21 +1019,24 @@ func TestReservedFreePurchasing(t *testing.T) {
 		wantOwned(t, auction, sim.Acc(1).From, 0, 0)
 	})
 
+	// Mint some free purchases before the normal sale starts
+	sim.Must(t, "PurchaseFreeOfCharge(freeQuota)")(auction.PurchaseFreeOfCharge(sim.Acc(0), sim.Acc(0).From, big.NewInt(1)))
+
 	t.Run("reserved free quota honoured", func(t *testing.T) {
 		n := big.NewInt(totalInventory - freeQuota)
 		sim.Must(t, "Buy(totalInventory - freeQuota)")(auction.Buy(sim.WithValueFrom(0, n), sim.Acc(0).From, n))
 
 		if diff := revert.SoldOut.Diff(
-			auction.Buy(sim.WithValueFrom(0, big.NewInt(1)), sim.Acc(0).From, big.NewInt(1)),
+			auction.Buy(sim.WithValueFrom(1, big.NewInt(1)), sim.Acc(1).From, big.NewInt(1)),
 		); diff != "" {
 			t.Errorf("After Buy(totalInventory - freeQuota); Buy(1) %s", diff)
 		}
 
-		wantOwned(t, auction, sim.Acc(0).From, totalInventory-freeQuota, 0)
+		wantOwned(t, auction, sim.Acc(0).From, totalInventory-freeQuota+1, 1)
```

Comment:
I think we should split out the accounts receiving free and purchased ones so it's very clear what's happening. I felt a tad uncomfortable with the tests that I wrote here and should have listened to my gut!

Perhaps define recipients and do something like this?

```Go
const (
  deployer = iota
  rcvFree
  buyer
)

…PurchaseFreeOfCharge(sim.Acc(deployer), sim.Acc(rcvFree).From, big.NewInt(1)))

…Buy(sim.WithValueFrom(buyer, big.NewInt(1)), sim.Acc(buyer).From, big.NewINt(1))

wantOwned(…, sim.Acc(rcvFree).From, freeQuota)
wantOwned(…, sim.Acc(buyer).From, freeQuota)
```

Can you please also add a `TestIssue8Regression()` with nothing but the exact steps to reproduce? Perhaps inventory of 20 and quota of 3; get 3 free then 14 paid then the remaining 3 paid. Pre fix only the last will fail, but post fix none will. The problem with the tests that I wrote are that they do too much so need to be more focussed.